#!/usr/bin/env python
PACKAGE = "image_utils"
NS = "General"
param_file = "general_params.yaml"

from dynamic_reconfigure.parameter_generator_catkin import *
import yaml
import os

Dir = os.path.dirname(os.path.realpath(__file__))

params = {}
with open(Dir+'/'+param_file, 'r') as stream:
  try:
    params = yaml.load(stream)
  except yaml.YAMLERROR as e:
    print e

gen = ParameterGenerator()

size_enum = gen.enum([gen.const("240p",   int_t,  240, "(320,240)"),
                      gen.const("360p",   int_t,  360, "(480,360)"),
                      gen.const("480p",   int_t,  480, "(640,480)"),
                      gen.const("720p",   int_t,  720, "(1280,720)"),
                      gen.const("1080p",  int_t,  1080, "(1920,1080)")], "Set size")

# Level 0
p = params['image']
gen.add("im_scale",          double_t,      0,   "Scale",    p['scale'],    0.5,     4)
gen.add("im_size",              int_t,      1,   "Size",     p['rgb_resolution'][1],    240,  1080, 
        edit_method=size_enum)

exit(gen.generate(PACKAGE, NS, "GeneralRQT"))
